developer_workflows.md

```diff
@@ -0,0 +1,50 @@
+# Developer Workflows
+
+When working with Git most teams use one of two different workflows depending on their preference.
+
+- [Feature branch workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) (also known as [GitHub flow](https://guides.github.com/introduction/flow/))
+- [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)
+
+Though FBW is the simpler of the two, it should be used by teams that are confident whenever they change the **master** branch it still remains ready for production ([Master is always shippable](https://github.com/DEFRA/dst-guides/blob/master/principles/master_is_always_shippable.md)).
+
+Gitflow is useful for teams that like the space to combine multiple features, or are working on a large feature, before then shipping to production.
+
+## Feature branch workflow (GitHub flow)
+
+<img src="githubflow.png" alt="GitHub flow" style="width: 600px;"/>
+<sub><https://guides.github.com/activities/hello-world/branching.png></sub>
+
+Using this workflow we only use two branches
+
+- **Master** is the main branch developers work from. It represents the latest version of the code. The key principle is that master is always production ready. Anything merged in needs to have been peer reviewed, passed by CI, and ready for release.
+
+- A **Feature** branch is started each time we want to add to, update or fix something in the code. We branch off Master when creating the feature, and once complete merge it back in
+
+## All branches
+
+In all cases when the branch is created it should be pushed up to the **origin** repo and a pull request created as per the [pull request](/process/pull-request) process.
+
+## Gitflow
+
+<img src="gitflow.png" alt="Git flow" style="width: 600px;"/>
+<sub><https://www.atlassian.com/git/images/tutorials/collaborating/comparing-workflows/gitflow-workflow/05.svg></sub>
+
+Using this workflow it means our branches have specific uses
+
+- **Master** is the version of code that is in production (see the principle [Master is always shippable](https://github.com/DEFRA/dst-guides/blob/master/principles/master_is_always_shippable.md))
+
+- We create a **Hotfix** branch when we need to make a change to production code because of a critical error. When finished we merge the change back into Master, but also Develop
+
+- **Develop** is the main branch the developers work from. It represents the current version of the code, including new features we’ve completed but not yet released
+
```

Comment:
Do/should we make this the default branch?

Reply:
That's a good call. If you are using gitflow you should be making `develop` your default branch in GitHub. I'll update the description with that note.